dev-docs/RFCs/v4.0.0/names_for_concepts_types_functions.md

```diff
@@ -0,0 +1,131 @@
+| `getPentagonIndexes`          | `getPentagons`        |                   |
+| `h3GetBaseCell`               | `getBaseCellNumber`   |                   |
+
+todo: Do we have general guidance on when to use a `get` prefix with a function and when not?
+
+
+## H3 Grid Functions
+
+| Current name |     Proposed name      |
+|--------------|------------------------|
+| `h3Distance` | `gridDistance`         |
+| `h3Line`     | `gridPathCell`         |
+| *DNE*        | `gridPathEdge`         |
+| *DNE*        | `gridPathDirectedEdge` |
+    
+todo: do we not use the `get` prefix for these functions? When to use vs. not use?
+
+## H3 Edge Types
+
+Instead of `UnidirectionalEdge`, use the term `DirectedEdge`.
+
+For a future undirected edge mode, use the term `Edge`.
+
+Function name changes:
+
+|                  Current name                 |        Proposed name         |
+|-----------------------------------------------|------------------------------|
+| `getH3UnidirectionalEdge`                     | `getDirectedEdge`            |
+| `h3UnidirectionalEdgeIsValid`                 | `isValidDirectedEdge`        |
+| `getOriginH3IndexFromUnidirectionalEdge`      | `getDirectedEdgeOrigin`      |
+| `getDestinationH3IndexFromUnidirectionalEdge` | `getDirectedEdgeDestination` |
+| `getH3IndexesFromUnidirectionalEdge`          | `getDirectedEdgeCells`       |
+| `getH3UnidirectionalEdgesFromHexagon`         | `getDirectedEdgesFromCell`   |
+| `getH3UnidirectionalEdgeBoundary`             | `getDirectedEdgeBoundary`    |
+
+
+todo: do we want/need to use `get` to prefix all these functions?
+
+## Area/Length Functions
+
+|  Current name  |    Proposed name     |                   Notes                    |
+|----------------|----------------------|--------------------------------------------|
+| `hexAreaKm2`   | `hexAreaAvgKm2`      | todo: `hexAreaMaxKm2` and `hexAreaMinKm2`? |
+| `hexAreaM2`    | `hexAreaAvgM2`       | todo: add min/max version?                 |
+| `edgeLengthKm` | `hexEdgeLengthAvgKm` | todo: add min/max version?                 |
+| `edgeLengthM`  | `hexEdgeLengthAvgM`  | todo: add min/max version?                 |
+| *DNE*          | `pentagonAreaAvgKm2` | plus others                                |
+
+todo: Could we add functions with signatures like `float cellAreaKm2(H3Index h)` to compute the actual area of a specific H3 cell?
+
+
+## Cell Neighborhood Functions
+
+| Current name | Proposed name |     Notes      |
+|--------------|---------------|----------------|
+| `kRing`      | `disk`        | filled-in disk |
+| `hexRing`    | `ring`        | hollow ring    |
+
+
+todo: For the `kRing`, `hexRange`, `hexRing`, etc. family of functions, should we come up with some standard prefix or suffix to denote that the function will fail if it encounters a pentagon?
+
+I'm anticipating that, at least in the wrappers, we'd probably just expose users to the "works in all cases" version of the function.
```

Comment:
I don't think that's a valid assumption for the Java and Go bindings. Performance can matter on large datasets, especially if being done "online" in the execution time of a user's request.

Reply:
That's true. A better way to say that would be that we'd probably *direct* new users to the "always works" functions, and let the fast versions have a possibly more cumbersome name like `<func_name>_hexonly` or something.

The point is just to make it obvious to the end user when two functions do essentially the same thing, but one only works on hexagons.